.github/workflows/mypy.yml

```diff
@@ -24,7 +25,21 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  mypy:
+  mypy-cases-generator:
+    name: Run mypy on Tools/cases_generator/
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.x"
```

Comment:
Are there constraints on `x`? The cases generator uses `match`/`case` so it's got to be at least Python 3.10.

Reply:
Let's use 3.11. We need dog fooding

Reply:
`3.x` is "the latest stable version of Python":

https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#specifying-a-python-version

That's 3.11 now, and will be 3.12 shortly after its release.

So `3.x` would be fine to minimise churn on this line (for next time).

Reply:
> So 3.x would be fine to minimise churn on this line (for next time).

Good!